src/http/status.cr

```diff
@@ -0,0 +1,174 @@
+# methods to easily identify the type of response.
+enum HTTP::Status
+  CONTINUE                        = 100
+  SWITCHING_PROTOCOLS             = 101
+  PROCESSING                      = 102
+  EARLY_HINTS                     = 103
+  OK                              = 200
+  CREATED                         = 201
+  ACCEPTED                        = 202
+  NON_AUTHORITATIVE_INFORMATION   = 203
+  NO_CONTENT                      = 204
+  RESET_CONTENT                   = 205
+  PARTIAL_CONTENT                 = 206
+  MULTI_STATUS                    = 207
+  ALREADY_REPORTED                = 208
+  IM_USED                         = 226
+  MULTIPLE_CHOICES                = 300
+  MOVED_PERMANENTLY               = 301
+  FOUND                           = 302
+  SEE_OTHER                       = 303
+  NOT_MODIFIED                    = 304
+  USE_PROXY                       = 305
+  SWITCH_PROXY                    = 306
+  TEMPORARY_REDIRECT              = 307
+  PERMANENT_REDIRECT              = 308
+  BAD_REQUEST                     = 400
+  UNAUTHORIZED                    = 401
+  PAYMENT_REQUIRED                = 402
+  FORBIDDEN                       = 403
+  NOT_FOUND                       = 404
+  METHOD_NOT_ALLOWED              = 405
+  NOT_ACCEPTABLE                  = 406
+  PROXY_AUTHENTICATION_REQUIRED   = 407
+  REQUEST_TIMEOUT                 = 408
+  CONFLICT                        = 409
+  GONE                            = 410
+  LENGTH_REQUIRED                 = 411
+  PRECONDITION_FAILED             = 412
+  PAYLOAD_TOO_LARGE               = 413
+  URI_TOO_LONG                    = 414
+  UNSUPPORTED_MEDIA_TYPE          = 415
+  RANGE_NOT_SATISFIABLE           = 416
+  EXPECTATION_FAILED              = 417
+  IM_A_TEAPOT                     = 418
+  MISDIRECTED_REQUEST             = 421
+  UNPROCESSABLE_ENTITY            = 422
+  LOCKED                          = 423
+  FAILED_DEPENDENCY               = 424
+  UPGRADE_REQUIRED                = 426
+  PRECONDITION_REQUIRED           = 428
+  TOO_MANY_REQUESTS               = 429
+  REQUEST_HEADER_FIELDS_TOO_LARGE = 431
+  UNAVAILABLE_FOR_LEGAL_REASONS   = 451
+  INTERNAL_SERVER_ERROR           = 500
+  NOT_IMPLEMENTED                 = 501
+  BAD_GATEWAY                     = 502
+  SERVICE_UNAVAILABLE             = 503
+  GATEWAY_TIMEOUT                 = 504
+  HTTP_VERSION_NOT_SUPPORTED      = 505
+  VARIANT_ALSO_NEGOTIATES         = 506
+  INSUFFICIENT_STORAGE            = 507
+  LOOP_DETECTED                   = 508
+  NOT_EXTENDED                    = 510
+  NETWORK_AUTHENTICATION_REQUIRED = 511
+
+  # Create a new status instance with the given status code, or raise an
+  # error if the status code given is not inside 100..999.
+  def self.from_code(status_code : Int32)
+    raise ArgumentError.new("Invalid HTTP status code: #{status_code}") unless 100 <= status_code <= 999
+    new(status_code)
+  end
+
+  def code
+    value
```

Comment:
I added it just for convenience because it read a little better, and it doesn't require the person using it to know it's an enum. Originally I didn't use it anywhere else - I left it just as a helpful addition, but another comment on the PR asked that I use `#code` everywhere.

Reply:
It's an alias but only because we can't remove `value`.

Reply:
Therefore:
- internally, `value` can be used because we know that's the enum,'s value and this avoid performance penalties (despite being minor) 
- `#code` has to be documented with comments

Reply:
Sure really minor, doesn't it calls `#code` that calls `#value`?

Reply:
Just asking what the gain of using a new method internally, we clearly see we are inside an Enum doesn't it?

Reply:
No, there's no perfomance penalty at all. There would eventually be one when you use Ruby but Crystal is *compiled* and efficient. As far as I know for example this:
```cr
def method1
  123
end
def method2
  method1
end
method2()
```
simply becomes
```cr
123
```
at compile-time.

Reply:
oh ok, same for `@ivar` and `ivar` I guess?

Reply:
Yes, the compiler (LLVM, but in some cases Crystal) inlines stuff. **A lot**.

Reply:
Thanks for the explanations.
Another good point of having this method is for documentation.